src/server/oauth.ts

```diff
@@ -236,30 +271,24 @@ export async function handleOAuthCallback(
       updatedCookies.push(result.updatedCookie);
       nonce = result.nonce;
     }
-    const result = await o.processAuthorizationCodeOpenIDResponse(
+    const result = await o.processAuthorizationCodeResponse(
       server,
       client,
       codeGrantResponse,
-      nonce ?? o.expectNoNonce,
+      {
+        expectedNonce: nonce ?? o.expectNoNonce,
+        requireIdToken: true,
```

Comment:
Do we understand what this does / why it's safe to change here?

Reply:
`oauth4webapi` 3.0.0 changed the API and replaced a couple functions with this one.

> processAuthorizationCodeOpenIDResponse() method was removed in favour of processAuthorizationCodeResponse()
> processAuthorizationCodeOAuth2Response() method was removed in favour of processAuthorizationCodeResponse()

https://github.com/panva/oauth4webapi/releases/tag/v3.0.0

These extra args passed here are for OIDC - Auth.js does it in a similar way:

https://github.com/nextauthjs/next-auth/blob/1c9bcdd0c4538a852f8d2b2b7c60eb962e3a50eb/packages/core/src/lib/actions/callback/oauth/callback.ts#L215-L223